.github/workflows/git-commit-checks.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import git
+import json
+import argparse
+
+GITHUB_WORKSPACE = os.environ.get('GITHUB_WORKSPACE')
+GITHUB_SHA       = os.environ.get('GITHUB_SHA')
+GITHUB_BASE_REF  = os.environ.get('GITHUB_BASE_REF')
+
+# Sanity check
+if GITHUB_WORKSPACE is None or GITHUB_SHA is None or GITHUB_BASE_REF is None:
+    print("Error: this script is designed to run as a Github Action")
+    exit(1)
+
+#----------------------------------------------------------------------------
+
+def print_result(good_list, bad_list, skipped_list):
+    def _print_list(msg, items, prefix=""):
+        print(msg)
+        for item in items:
+            print(f"{prefix}- {item}")
+
+    passed = True
+    if len(good_list):
```

Comment:
Do we really care about the list of good commits?

Reply:
I agree: It isn't critical to the required functionality.  Especially in automated actions, I tend to default to printing out more information than less so that you can tell that the automation actually did what it was supposed to do.  In this case, "I saw this commit, and I declared that it was good."  My rationale was that this covered cases like inadvertent tail drops and the like (e.g., make sure the script didn't parse *some* of the commits, but not *all* of the commits).

That being said, I don't have a strong feeling about this particular functionality.  If you don't think it should be there, I can delete it.